src/transformers/models/video_llava/modeling_video_llava.py

```diff
@@ -0,0 +1,667 @@
+            raise ValueError(
+                f"The input provided to the model are wrong. The number of {visual_type} tokens is {torch.sum(special_image_token_mask)} while"
+                f" the number of {visual_type} given to the model is {num_images}. This prevents correct indexing and breaks batch generation."
+            )
+
+        final_embedding[image_to_overwrite] = visual_features.contiguous().reshape(-1, embed_dim).to(target_device)
+        final_attention_mask |= image_to_overwrite
+        position_ids = (final_attention_mask.cumsum(-1) - 1).masked_fill_((final_attention_mask == 0), 1)
+
+        return final_embedding, final_attention_mask, final_labels, position_ids, final_input_ids
+
+    def _get_vision_features(
+        self,
+        pixel_values_images: Optional[torch.FloatTensor] = None,
+        pixel_values_videos: Optional[torch.FloatTensor] = None,
+        vision_feature_layer: Optional[int] = None,
+        vision_feature_select_strategy: Optional[str] = None,
+    ) -> Union[Tuple, BaseModelOutputWithPooling]:
+        if pixel_values_images is None and pixel_values_videos is None:
+            raise ValueError("You have to specify `pixel_values_images` or `pixel_values_videos`")
+
+        # videos do not need to select features and it's always "full" (as it is done in the orig implementation)
+        if pixel_values_videos is not None:
+            batch_size_vid, num_frames, channels, height, width = pixel_values_videos.shape
+            pixel_values = pixel_values_videos.reshape(batch_size_vid * num_frames, channels, height, width)
+            video_outputs = self.video_tower(
+                pixel_values,
+                output_hidden_states=True,
+            )
+            video_outputs = video_outputs.hidden_states[vision_feature_layer].squeeze(1)
+        else:
+            video_outputs = None
+
+        if pixel_values_images is not None:
+            image_outputs = self.image_tower(
+                pixel_values_images,
+                output_hidden_states=True,
+            )
+
+            image_outputs = image_outputs.hidden_states[vision_feature_layer].squeeze(1)
+
+            if vision_feature_select_strategy == "default":
+                image_outputs = image_outputs[:, 1:]
+            elif vision_feature_select_strategy == "full":
+                image_outputs = image_outputs
+            else:
+                raise ValueError(f"Unexpected select feature strategy: {self.config.vision_feature_select_strategy}")
+        else:
+            image_outputs = None
+
+        return image_outputs, video_outputs
+
+    @add_start_docstrings_to_model_forward(VIDEO_LLAVA_INPUTS_DOCSTRING)
+    @replace_return_docstrings(output_type=VideoLlavaCausalLMOutputWithPast, config_class=_CONFIG_FOR_DOC)
+    def forward(
+        self,
+        input_ids: torch.LongTensor = None,
+        pixel_values_images: torch.FloatTensor = None,
+        pixel_values_videos: torch.FloatTensor = None,
```

Comment:
It's not ideal having to change the standard input type names. I _think_ this is OK, as we have e.g. `input_ids_query`, `input_ids_doc` for retribert